### PR TITLE
#165413064  add password reset functionality

### PR DESCRIPTION
**Pull request by Niyitangasam**
Description
=======

This  PR adds resetting functionality where it provides two components, one of sending an email with instructions and other for changing password

Type of change
=======
N/A

How has it been tested
=======
It can be tested by accessing below routes:
``/reset``
``/completeReset/:token``

Checklist:
=======
- Add components for sending an email and changing password.
- Add necessary HTML and scss to the page

Pivotal tracker story ID
=======
[#165413064](https://www.pivotaltracker.com/n/projects/2326984/stories/165413064)

Screenshots
=======

Screen Shot 2019-07-01 at 02 48 09

Screen Shot 2019-07-01 at 02 49 01

Screen Shot 2019-07-01 at 02 50 24

### CODE REVIEW

**Review by AJAkimana — APPROVED** [review]
Great work @Niyitangasam

**Review by ericnyirimana — APPROVED** [review]

File: src/reducers/index.js [hunk 1]
```diff
@@ -1,5 +1,7 @@
 import { combineReducers } from "redux";
 import testReduscer from './testReducer';
+import resetPasswordReducer from './resetPasswordReducer';
 export default combineReducers({
-  dummyState: testReduscer
+  dummyState: testReduscer,
```
Comment (macphilips) [inline src/reducers/index.js]: remove this

File: .travis.yml [hunk 2]
```diff
@@ -4,6 +4,10 @@ language:
 node_js: 
   "10.15.3"
 
+dist: xenial
+services:
+  - xvfb
```
Comment (macphilips) [inline .travis.yml]: why is this guy here?

File: cypress/integration/reset.spec.js [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+        cy.visit('/completReset/token');
+        cy.get("#newPassword")
+          .type("Password@123")
+          .should("have.value", "Password@123");
+        cy.get("#confirmNewPassword")
+          .type("Password@123")
+          .should("have.value", "Password@123");
+        cy.get('#btn-send').click();
+      });
+  });
```
Comment (macphilips) [inline cypress/integration/reset.spec.js]: add new line

File: .travis.yml (hunk 2 above)
Comment (Niyitangasam) [reply .travis.yml]: It is a service called virtual framebuffer that help us to run end to end tests on Travis.

File: .travis.yml [hunk 4]
```diff
@@ -2,6 +2,10 @@ language: node_js
 
 node_js: '10.15.3'
 
+dist: xenial
```
Comment (macphilips) [inline .travis.yml:5]: duplicate definition

File: cypress/integration/reset.spec.js [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+        cy.visit('/completReset/token');
+        cy.get("#newPassword")
+          .type("Password@123")
+          .should("have.value", "Password@123");
+        cy.get("#confirmNewPassword")
+          .type("Password@123")
+          .should("have.value", "Password@123");
+        cy.get('#btn-send').click();
+        cy.wait(1000);
+        cy.get('.Toastify').should('not.be.visible');
```
Comment (macphilips) [inline cypress/integration/reset.spec.js:21]: From the description of this test, can't see any `should` or `except` assertion for the displayed error message

File: package.json [hunk 6]
```diff
@@ -5,7 +5,7 @@
   "main": "index.js",
   "scripts": {
     "build": "NODE_ENV=production babel-node ./node_modules/webpack/bin/webpack --config webpack/prod.config.js",
-    "start": "NODE_ENV=development babel-node ./node_modules/webpack-dev-server/bin/webpack-dev-server --config webpack/dev.config.js",
+    "start": "NODE_ENV=development babel-node ./node_modules/webpack-dev-server/bin/webpack-dev-server --config webpack/dev.config.js --open",
```
Comment (macphilips) [inline package.json]: What is the open flag for?

Comment (Niyitangasam) [reply package.json]: The open flag starts the app automatically in the browser without typing URL in browser as create react app does when you run the app.

File: src/reducers/index.js (hunk 1 above)
Comment (Niyitangasam) [reply src/reducers/index.js]: Removed in sign up PR

**Pull request merged**